core/flyout_base.js

```diff
@@ -612,11 +612,17 @@ Blockly.Flyout.prototype.createBlock_ = function(blockXml) {
  * @private
  */
 Blockly.Flyout.prototype.getBlockXml_ = function(blockInfo) {
-  var blockXml = blockInfo['blockxml'];
-  if (blockXml) {
+  var blockXml = null;
+  // All blockInfo will have type, so check for blockxml first.
+  if (blockInfo['blockxml']) {
     blockXml = Blockly.Xml.textToDom(blockInfo['blockxml']);
+  } else if (blockInfo['type']) {
```

Comment:
We should update `Blockly.utils.toolbox.Block` to include type, and make blockxml optional.